### ISSUE TITLE
No feed file generated in multilingual sites

### ISSUE THREAD

**Issue by theowenyoung** [Keats/gutenberg#1896]
# Bug Report

## Environment

Zola version: 0.15.3

## Expected Behavior

`en` as my second language, I configured the taxonomies `feed: true`, 

```toml
[languages.en]
taxonomies = [
  {name = "tags", feed = true, paginate_by = 20, lang = "en"},
  {name = "categories", feed = true, paginate_by = 20, lang = "en"},
]
```

When I build, there should be `public/en/categories/{name}/atom.xml` generated

## Current Behavior
There are no `public/en/categories/{name}/atom.xml` generated, they are only generated in default language, `public/categories/{name}/atom.xml`

## Step to reproduce

With this multilingual configure:

```toml
[languages.en]
taxonomies = [
  {name = "tags", feed = true, paginate_by = 20, lang = "en"},
  {name = "categories", feed = true, paginate_by = 20, lang = "en"},
]
```

**Comment by Keats** [comment]
I couldn't reproduce it in the `next` branch but I did find and fix a bug (https://github.com/getzola/zola/pull/1741/commits/4da81eb17e7995e09e6ede75d3e205a0c5f1cbd2) related to the position of the taxonomies feeds.

**Comment by theowenyoung** [comment]
Well, I made a minimal example repo for this issue: <https://github.com/theowenyoung/zola-reproduce-example>

```bash
git clone git@github.com:theowenyoung/zola-reproduce-example.git
zola build
```

You will see `public/categories/random/atom.xml`, but not for `public/zh/categories/random/atom.xml`.


[Issue track](https://github.com/getzola/zola/issues/1896)

Related config:

```toml
# The URL the site will be built for
base_url = "https://example.com"

# Whether to automatically compile all Sass files in the sass directory
compile_sass = false

# Whether to build a search index to be used later on by a JavaScript library
build_search_index = false
generate_feed = true
taxonomies = [
  # You can enable/disable RSS
  {name = "categories", feed = true},
  {name = "tags", feed = true},
]
theme = "after-dark"
[languages.zh]
taxonomies = [
  # You can enable/disable RSS
  {name = "categories", feed = true},
  {name = "tags", feed = true},
]
[markdown]
# Whether to do syntax highlighting
# Theme can be customised by setting the `highlight_theme` variable to a theme supported by Zola
highlight_code = false

[extra]
# Put all your custom variables here
```

**Comment by Keats** [comment]
Thanks for the repro! It's fixed in the `next` branch, I see the file now. Languages other than the default lang were put in the wrong place and being overwritten.

**Comment by theowenyoung** [comment]
Thanks, I try the next branch, and it works! But the build time seems to be a lot more than the current stable version, is that expected?

**next branch**: 6.2s
**stable**: 2.6s

```bash
➜  blog git:(main) ✗ ../inbox/zola/target/release/zola build
Building site...
Checking all internal links with anchors.
> Successfully checked 3 internal link(s) with anchors.
-> Creating 134 pages (8 orphan) and 10 sections
Done in 6.2s.

➜  blog git:(main) ✗ make build
./bin/zola build
Building site...
Checking all internal links with anchors.
> Successfully checked 3 internal link(s) with anchors.
-> Creating 134 pages (8 orphan) and 10 sections
Done in 2.6s.
```

**Comment by theowenyoung** [comment]
Another thing is that the `next` branch also not generate `public/zh/atom.xml`, is it a bug?

**Comment by Keats** [comment]
Did you build it with `cargo build --release`? Build time should not have changed.

> Another thing is that the next branch also not generate public/zh/atom.xml, is it a bug?

I'll have a look

**Comment by Keats** [comment]
The repro is missing `generate_feed = true` in the zh language to generatel the overall feed

**Comment by theowenyoung** [comment]
> Did you build it with cargo build --release? Build time should not have changed.

Yes, I built it with `--release`,   here is my blog repo action: <https://github.com/theowenyoung/blog/runs/6942525815?check_suite_focus=true>


here is before: <https://github.com/theowenyoung/blog/runs/6833155916?check_suite_focus=true>

but anyway, it's not annoying me, cause it absolute blaze fast even now. 

> The repro is missing generate_feed = true in the zh language to generatel the overall feed

It works now, thanks very much!

**Comment by Keats** [comment]
FYI on my machine for the `next` branch:

```
~/C/p/blog (main|✔) $ hyperfine -r 10 "zola build"
Benchmark 1: zola build
  Time (mean ± σ):      2.375 s ±  0.122 s    [User: 11.125 s, System: 1.569 s]
  Range (min … max):    2.243 s …  2.589 s    10 runs
```

and for 0.15.3:

```
~/C/p/blog (main|✔) $ hyperfine -r 10 "zola build"
Benchmark 1: zola build
  Time (mean ± σ):      1.463 s ±  0.239 s    [User: 4.709 s, System: 0.692 s]
  Range (min … max):    1.142 s …  1.800 s    10 runs
```
So there is a very significant slowdown (>40%) for this site on the `next` branch, I'll investigate that. Closing this issue and I'll open another one for the slowdown.

**Comment by c-git** [comment]
# TLDR

It is not slower on my sites. It's actually faster.

- - -

# Detailed version

Hi, I tried it on my 3 of my sites (they are all small). For me the one on next branch was faster than the snap version. (At the end I did one more after compiling the master branch to see if the issue was the snap).

## Site 1

```
$ hyperfine -r 10 "/snap/bin/zola build"
Benchmark 1: /snap/bin/zola build
  Time (mean ± σ):     120.2 ms ±   1.6 ms    [User: 197.5 ms, System: 37.9 ms]
  Range (min … max):   117.3 ms … 122.6 ms    10 runs
```

```
$ hyperfine -r 10 "/tmp/zola/target/release/zola build"
Benchmark 1: /tmp/zola/target/release/zola build
  Time (mean ± σ):      95.0 ms ±   5.6 ms    [User: 284.7 ms, System: 37.3 ms]
  Range (min … max):    89.1 ms … 104.3 ms    10 runs
```

## Site 2

```
$ hyperfine -r 10 "/snap/bin/zola build"
Benchmark 1: /snap/bin/zola build
  Time (mean ± σ):      35.8 ms ±   1.3 ms    [User: 16.3 ms, System: 22.6 ms]
  Range (min … max):    34.7 ms …  38.6 ms    10 runs
```

```
$ hyperfine -r 10 "/tmp/zola/target/release/zola build"
Benchmark 1: /tmp/zola/target/release/zola build
  Time (mean ± σ):      14.5 ms ±   1.8 ms    [User: 6.8 ms, System: 9.4 ms]
  Range (min … max):    12.6 ms …  18.6 ms    10 runs
```


## Site 3

```
$ hyperfine -r 10 "/snap/bin/zola build"
Benchmark 1: /snap/bin/zola build
  Time (mean ± σ):      77.5 ms ±   1.0 ms    [User: 72.7 ms, System: 24.3 ms]
  Range (min … max):    76.2 ms …  79.1 ms    10 runs
```

```  
$hyperfine -r 10 "/tmp/zola/target/release/zola build"
Benchmark 1: /tmp/zola/target/release/zola build
  Time (mean ± σ):      38.2 ms ±   1.6 ms    [User: 48.6 ms, System: 14.7 ms]
  Range (min … max):    35.6 ms …  40.4 ms    10 runs
```

I wondered if it was because of the snap so I compiled the master branch to compare and it was faster than the snap but not as fast as the next branch. Below is the output on  he last site.

```
$ hyperfine -r 10 "/tmp/zola/target/release/zola build"
Benchmark 1: /tmp/zola/target/release/zola build
  Time (mean ± σ):      49.9 ms ±   2.0 ms    [User: 54.2 ms, System: 14.3 ms]
  Range (min … max):    47.8 ms …  53.5 ms    10 runs
```

**Comment by Keats** [comment]
Thanks for checking!
